test/compress/dead-code.js

```diff
@@ -68,10 +68,9 @@ dead_code_constant_boolean_should_warn_more: {
         side_effects : true,
     };
     input: {
-        while (!((foo && bar) || (x + "0"))) {
+        while (!(foo || (x + "0"))) {
             console.log("unreachable");
             var foo;
-            function bar() {}
```

Comment:
Does `minify()` need a top level `ecma` option to specify whether to parse/compress in `ecma: 5` or `ecma: 6` mode? `output.js` already has `ecma: 6` support to print ES6 properties.